clientv3/kv.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv3
+
+type KV interface {
+	// PUT puts a key-value pair into etcd.
+	// Note that key,value can be plain bytes array and string is
+	// an immutable representation of that bytes array.
+	// To get a string of bytes, do string([]byte(0x10, 0x20)).
+	Put(key, val string) (PutResponse, error)
```

Comment:
Are these going to be replaced with something closer to the current `etcd/client` where you have a single `Do(Request) error` method, and then pass in the different operations as structs? Ex:

```
type KeyValueOp interface {
    Request() *Request
}

type KV interface {
     Do(*Request) (*Response, error)
}

type PutRequest struct {}
func (r *PutRequest) Request() (*Request, error) { 
   ...
}
``` 

If so, you could then just re-use these structs as inputs to the `Then/Else` methods in the TX interface.